modules/frontend/searchsharding.go

```diff
@@ -276,6 +299,17 @@ func (s searchSharder) RoundTrip(r *http.Request) (*http.Response, error) {
 	}, nil
 }
 
+// recordSearchMetrics records throughput and SLO metrics
+func (s *searchSharder) recordMetrics(tenantID string, status int, throughput float64, reqTime time.Duration) {
+	sts := strconv.Itoa(status)
+	searchThroughput.WithLabelValues(tenantID, sts).Observe(throughput)
+
+	// query is within SLO if query returned within DurationSLO seconds OR processed ThroughputSLO bytes/s data
+	if reqTime < s.sloCfg.DurationSLO || throughput > s.sloCfg.ThroughputSLO {
+		sloSearchPerTenant.WithLabelValues(tenantID, sts).Inc()
```

Comment:
i find this metric a little confusing. the name is `query_frontend_search_queries_within_slo_total` but
- if the status >= 500 then the that is not a succesful query
- this metric is used for both search and find trace by id

maybe there's a more accurate name? i'd also consider removing status and adding the endpoint hit.

Reply:
only recording this in case of 200, and removed status from the metric, metric is renamed and added op label for search query types.